src/util/strencodings.cpp

```diff
@@ -107,25 +107,35 @@ std::vector<unsigned char> ParseHex(const std::string& str)
     return ParseHex(str.c_str());
 }
 
-void SplitHostPort(std::string in, uint16_t& portOut, std::string& hostOut)
+bool SplitHostPort(std::string in, uint16_t& portOut, std::string& hostOut)
 {
+    bool valid = false;
     size_t colon = in.find_last_of(':');
     // if a : is found, and it either follows a [...], or no other : is in the string, treat it as port separator
     bool fHaveColon = colon != in.npos;
     bool fBracketed = fHaveColon && (in[0] == '[' && in[colon - 1] == ']'); // if there is a colon, and in[0]=='[', colon is not 0, so in[colon-1] is safe
     bool fMultiColon = fHaveColon && (in.find_last_of(':', colon - 1) != in.npos);
     if (fHaveColon && (colon == 0 || fBracketed || !fMultiColon)) {
-        uint16_t n;
-        if (ParseUInt16(in.substr(colon + 1), &n)) {
+        if (!in.substr(colon + 1).empty()) {
+            uint16_t n;
+            if (ParseUInt16(in.substr(colon+1), &n)) {
+              portOut = n;
+              valid = (portOut != 0);
+            }
             in = in.substr(0, colon);
-            portOut = n;
+        } else {
+          valid = true;
         }
+    } else {
+      valid = true;
```

Comment:
IMO we should be setting `portOut` to `0` here explicitly. Maybe make it a bugfix commit before your others.

Reply:
The way this has been used up to now makes for neat code:
https://github.com/bitcoin/bitcoin/blob/25290071c434638e2719a99784572deef44542ad/src/netbase.cpp#L203-L205
Don't you think it's a bit neater this way?

Reply:
Yes, I agree.